src/swarm-lang/Swarm/Language/Types.hs

```diff
@@ -423,41 +432,110 @@ instance (UnchainableFun t, PrettyPrec t, SubstRec t) => PrettyPrec (TypeF t) wh
 ------------------------------------------------------------
 
 class Typical t where
-  foldT :: (TypeF t -> t) -> t -> t
+  -- Fold a type into a summary value of some type @a@, given an
+  -- "empty" value of type @a@ (for use at e.g. Pure nodes in a UType)
+  foldT :: a -> (TypeF a -> a) -> t -> a
+
+  -- Refold a type into another type
+  refoldT :: (TypeF t -> t) -> t -> t
```

Comment:
This looks truly magical. 🪄

Reply:
What seems magical about it?  Is there additional documentation that would be helpful?